openhtf/__init__.py

```diff
@@ -394,6 +394,33 @@ def with_args(self, **kwargs):
     new_info.measurements = [m.with_args(**kwargs) for m in self.measurements]
     return new_info
 
+  def with_plugs(self, **subplugs):
+    """Substitute plugs for placeholders for this phase."""
+    # Make a copy so we can have multiple of the same phase with different plugs
+    # in the same test.
+    new_info = mutablerecords.CopyRecord(self)
+    
+    for substitute_name, substitute_class in subplugs.iteritems():
+      # self.plugs contains PhasePlug so enumerate and unpack
+      for i, original_plug in enumerate(self.plugs):
```

Comment:
O(n^2) :'(  How about iterating over self.plugs and removing keys from subplugs as you find them, then at the end any keys left in subplugs are ones for which you couldn't find a valid placeholder.